src/xtgeo/grid3d/_grdecl_grid.py

```diff
@@ -330,12 +330,16 @@ def _to_grdecl_file(self, filename):
             for kw, values in keywords:
                 if values is None:
                     continue
-                filestream.write(kw)
-                if values is not None:
-                    filestream.write("\n")
-                    for value in values:
-                        filestream.write(" ")
-                        filestream.write(str(value))
+                filestream.write(f"{kw}\n")
+                lcount = 0
+                for value in values:
+                    lcount += 1
+                    filestream.write(f" {value}")
+
+                    if kw in ("COORD", "ZCORN", "ACTNUM") and lcount >= 6:
```

Comment:
Why only these three keywords? If this function is called with a different keyword, will the newline never be printed?

Reply:
Ah, I see that this lines can only be triggered by 8 selected keywords. But is the `in` test necessary, will the other keywords reach beyound 6 elements?

Reply:
They should not do, but I agree; will now use this for all (simplifies code)